### ISSUE TITLE
parallels-iso fails to SSH in if prepare_iso.sh run with custom username/password

### ISSUE THREAD

**Issue by wizonesolutions** [timsutton/osx-vm-templates#26]
Downloaded the OS X Yosemite installer from the App Store.

Ran the prepare script on it, specifying a custom user name and password.

Gave the same username and password to `packer build`.

Wait forever.

Got:

http://gyazo.com/600723ef92b34153c00a725618f0cc99

:o

**Comment by wizonesolutions** [comment]
Oh! I tried with default username/password, and it was able to log in. So maybe something with the prepare script.

BUT, I got this: http://gyazo.com/8b551562702d99e5fbc0d644f098e920

So it didn't finish writing the all-important box file :(

Far as I could see, I was not actually out of disk space. But I will try on another disk with a lot more. Also updating issue title to be more specific.

**Comment by timsutton** [comment]
Just did a test using vmware-iso and parallels-iso builders, seems to be working for me. Did you specify your user and password to both the prepare script and vars for `packer build`?

```
sudo prepare_iso/prepare_iso.sh \
-u other user \
-p other user \
/Applications/Install\ OS\ X\ Yosemite.app \
out
```

```
packer build \
-except=virtualbox-iso \
-var username=otheruser \
-var password=otheruser \
-var install_vagrant_keys=false \
-var iso_checksum=2052510dc67a8736c0c604f1ebdf5d01 \
-var iso_url=../out/OSX_InstallESD_10.10.1_14B25.dmg \
template.json
```

**Comment by wizonesolutions** [comment]
@timsutton I did! I wonder if I fat-fingered it somewhere, though, or if `prepare_iso.sh` is somewhat dependent on the order of arguments fed to it? I think I might have provided the path to the installation media first and `-u`/`-p` after.

I'll give it another run. It'll save me having to create a new user profile with Ansible.

**Comment by timsutton** [comment]
@wizonesolutions I didn't write the support for the `[-upi]` switches, but they're documented as being before the paths (which are required args), so I'm guessing it would not work if you put those switches after.

One of these days I will move this thing from Bash to Python to make it more robust and maintainable.

**Comment by wizonesolutions** [comment]
Interesting; this time it worked fine.

I took a look at my `.bash_history`, and in the failure case I was using the following incantation:

`sudo prepare_iso/prepare_iso.sh /Applications/Install\ OS\ X\ Yosemite.app out -u kevin -p secret`

I passed these to `packer build` in the same way I did today, which is why I'm led to believe that order of arguments to `prepare_iso.sh` matters. And honestly, the way you did it in your comment is the same as what the README says, so I don't have much of an excuse :)

Thanks for helping me out all the same. Closing since it's obviously user error.
